logging/log.go

```diff
@@ -0,0 +1,157 @@
+// Copyright © 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package logging
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+const (
+	// LogLevelError specified the logging level as: ERROR
+	LogLevelError = 1
+
+	// LogLevelWarning specified the logging level as: WARNING
+	LogLevelWarning = 2
+
+	// LogLevelInfo specified the logging level as: INFO
+	LogLevelInfo = 3
+
+	// LogLevelDebug specified the logging level as: DEBUG
+	LogLevelDebug = 4
+
+	// LogLevelVerbose specified the logging level as: VERBOSE
+	// This is the same as Debug, but without the repeat filtering
+	LogLevelVerbose = 5
+)
+
+var (
+	level      = LogLevelInfo
+	levelMap   = map[int]string{}
+	fileHandle *os.File
+	logging    bool
+	lineLast   string
+	lineCount  int
+)
+
+func init() {
+	levelMap[LogLevelError] = "LogLevelError"
+	levelMap[LogLevelWarning] = "LogLevelWarning"
+	levelMap[LogLevelInfo] = "LogLevelInfo"
+	levelMap[LogLevelDebug] = "LogLevelDebug"
+	levelMap[LogLevelVerbose] = "LogLevelVerbose"
+}
+
+// SetLogLevel sets the default logging level to l
+func SetLogLevel(l int) {
+	if l < LogLevelError {
+		level = LogLevelError
+		logTag("WRN", "Log Level '%d' too low, forcing to %s (%d)", l, levelMap[level], level)
+	} else if l > LogLevelVerbose {
+		level = LogLevelVerbose
+		logTag("WRN", "Log Level '%d' too high, forcing to %s (%d)", l, levelMap[level], level)
+	} else {
+		level = l
+		Debug("Log Level set to %s (%d)", levelMap[level], l)
+	}
+}
+
+// SetOutputFilename ... sets the default logging output to filename instead of stdout/stderr
+func SetOutputFilename(logFile string) (*os.File, error) {
```

Comment:
We should have a default log file "mixer-tools.log"

Reply:
Then should logging be always enabled by default?